File: pax-romana-service/pax-romana-service/Controllers/PlayerController.cs
using Microsoft.AspNetCore.Mvc;
using pax_romana_service.Dependencies;
using pax_romana_service.Models;

namespace pax_romana_service.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    public class PlayerController: ControllerBase
    {
        private readonly ILogger<PlayerController> _logger;
        private readonly Repository _repo;

        public PlayerController(ILogger<PlayerController> logger, Repository repo)
        {
            _logger = logger;
            _repo = repo;
        }

        /// <summary>
        /// Retrieves the player's data by their Id
        /// </summary>
        /// <param name="playerId"></param>
        /// <returns>A Player object</returns>
        [HttpGet("{playerId}")]
        public ActionResult<Player> GetPlayerById([FromRoute] int playerId)
        {
            try
            {
                if (playerId == 0)
                {
                    _logger.LogError("PlayerId is not valid: {playerId}", playerId);
                    return BadRequest($"Error when getting player data: PlayerId is not valid {playerId}");
                }
                _logger.LogInformation("Received request to get data for player: {playerId}", playerId);
                var player = _repo.GetPlayerById(playerId);
                return Ok(player);
            }
            catch (Exception ex)
            {
                _logger.LogError("Error when getting player data: {playerId}", playerId);
                return BadRequest($"Error when getting player data: {ex.Message}");
            }
        }

    }
}

File: pax-romana-ui/app/controllers/playerController.ts
import type { Player } from "~/models/player";
import { ApiConfig as api } from "~/ApiConfig";

/**
 * Retrieves player data by ID
 * @param playerId number
 * @returns Promise<Player>
 */
export async function getPlayerById(playerId: number): Promise<Player> {
  if (!playerId || playerId === 0) {
    throw new Error(`Invalid playerId: ${playerId}`);
  }

  try {
    const response = await fetch(`${api.player}/${playerId}`, {
      method: "GET",
      headers: {
        "Content-Type": "application/json",
      },
    });

    if (!response.ok) {
      const errorText = await response.text();
      throw new Error(`Error fetching player: ${errorText}`);
    }

    const data: Player = await response.json();
    return data;
  } catch (error) {
    console.error("Error fetching player data:", error);
    throw error;
  }
}